File: WidgetDesigners/Account/AccountDesigner.js
Type.registerNamespace("SitefinityWebApp.WidgetDesigners.Account");

SitefinityWebApp.WidgetDesigners.Account.AccountDesigner = function (element) {
    /* Initialize Message fields */
    this._message = null;
    this._viewtemplate = null;
    this._loginrequired = null;
    /* Calls the base constructor */
    SitefinityWebApp.WidgetDesigners.Account.AccountDesigner.initializeBase(this, [element]);
}

SitefinityWebApp.WidgetDesigners.Account.AccountDesigner.prototype = {
    /* --------------------------------- set up and tear down --------------------------------- */
    initialize: function () {
        /* Here you can attach to events or do other initialization */
        SitefinityWebApp.WidgetDesigners.Account.AccountDesigner.callBaseMethod(this, 'initialize');
    },
    dispose: function () {
        /* this is the place to unbind/dispose the event handlers created in the initialize method */
        SitefinityWebApp.WidgetDesigners.Account.AccountDesigner.callBaseMethod(this, 'dispose');
    },

    /* --------------------------------- public methods ---------------------------------- */

    findElement: function (id) {
        var result = jQuery(this.get_element()).find("#" + id).get(0);
        return result;
    },

    /* Called when the designer window gets opened and here is place to "bind" your designer to the control properties */
    refreshUI: function () {
        var controlData = this._propertyEditor.get_control().Settings; /* JavaScript clone of your control - all the control properties will be properties of the controlData too */

        /* RefreshUI Message */
        jQuery(this.get_message()).val(controlData.Message);
        //ViewTemplate
        jQuery(this.get_viewtemplate()).val(controlData.ViewTemplate);
        
        if(controlData.LoginRequired){ jQuery(this.get_loginrequired()).prop('checked', true); }
    },

    /* Called when the "Save" button is clicked. Here you can transfer the settings from the designer to the control */
    applyChanges: function () {
        var controlData = this._propertyEditor.get_control().Settings;

        /* ApplyChanges Message */
        controlData.Message = jQuery(this.get_message()).val();
        //ViewTemplate
        controlData.ViewTemplate = jQuery(this.get_viewtemplate()).val();
        controlData.LoginRequired = false;
        if(jQuery(this.get_loginrequired()).is(':checked')){controlData.LoginRequired = true;}
    },

    /* --------------------------------- event handlers ---------------------------------- */

    /* --------------------------------- private methods --------------------------------- */

    /* --------------------------------- properties -------------------------------------- */

    /* Message properties */
    get_message: function () { return this._message; }, 
    set_message: function (value) { this._message = value; },

    get_viewtemplate: function () { return this._viewtemplate; }, 
    set_viewtemplate: function (value) { this._viewtemplate = value; },


    get_loginrequired: function () { return this._loginrequired; }, 
    set_loginrequired: function (value) { this._loginrequired = value; }
}

SitefinityWebApp.WidgetDesigners.Account.AccountDesigner.registerClass('SitefinityWebApp.WidgetDesigners.Account.AccountDesigner', Telerik.Sitefinity.Web.UI.ControlDesign.ControlDesignerBase);

File: WidgetDesigners/Account/AccountDesigner.cs
using System;
using System.Linq;
using System.Web.UI;
using Telerik.Sitefinity.Web.UI;
using Telerik.Sitefinity.Web.UI.ControlDesign;
using System.Collections.Generic;
using System.Web.UI.WebControls;

namespace SitefinityWebApp.WidgetDesigners.Account
{
    /// <summary>
    /// Represents a designer for the <typeparamref name="SitefinityWebApp.Mvc.Controllers.AccountController"/> widget
    /// </summary>
    public class AccountDesigner : ControlDesignerBase
    {
        #region Properties
        /// <summary>
        /// Obsolete. Use LayoutTemplatePath instead.
        /// </summary>
        protected override string LayoutTemplateName
        {
            get
            {
                return string.Empty;
            }
        }

        /// <summary>
        /// Gets the layout template's relative or virtual path.
        /// </summary>
        public override string LayoutTemplatePath
        {
            get
            {
                if (string.IsNullOrEmpty(base.LayoutTemplatePath))
                    return AccountDesigner.layoutTemplatePath;
                return base.LayoutTemplatePath;
            }
            set
            {
                base.LayoutTemplatePath = value;
            }
        }

        protected override HtmlTextWriterTag TagKey
        {
            get
            {
                return HtmlTextWriterTag.Div;
            }
        }
        #endregion

        #region Control references
        /// <summary>
        /// Gets the control that is bound to the Message property
        /// </summary>
        protected virtual Control Message
        {
            get
            {
                return this.Container.GetControl<Control>("Message", true);
            }
        }
        /// <summary>
        /// Gets the control that is bound to the Message property
        /// </summary>
        protected virtual Control ViewTemplate
        {
            get
            {
                return this.Container.GetControl<Control>("ViewTemplate", true);
            }
        }
        protected virtual CheckBox LoginRequired
        {
            get
            {
                return this.Container.GetControl<CheckBox>("LoginRequired", true);
            }
        }
        #endregion

        #region Methods
        protected override void InitializeControls(Telerik.Sitefinity.Web.UI.GenericContainer container)
        {
            // Place your initialization logic here
        }
        #endregion

        #region IScriptControl implementation
        /// <summary>
        /// Gets a collection of script descriptors that represent ECMAScript (JavaScript) client components.
        /// </summary>
        public override System.Collections.Generic.IEnumerable<System.Web.UI.ScriptDescriptor> GetScriptDescriptors()
        {
            var scriptDescriptors = new List<ScriptDescriptor>(base.GetScriptDescriptors());
            var descriptor = (ScriptControlDescriptor)scriptDescriptors.Last();

            descriptor.AddElementProperty("message", this.Message.ClientID);
            descriptor.AddElementProperty("viewtemplate", this.ViewTemplate.ClientID);
            descriptor.AddElementProperty("loginrequired", this.LoginRequired.ClientID);

            return scriptDescriptors;
        }

        /// <summary>
        /// Gets a collection of ScriptReference objects that define script resources that the control requires.
        /// </summary>
        public override System.Collections.Generic.IEnumerable<System.Web.UI.ScriptReference> GetScriptReferences()
        {
            var scripts = new List<ScriptReference>(base.GetScriptReferences());
            scripts.Add(new ScriptReference(AccountDesigner.scriptReference));
            return scripts;
        }
        #endregion

        #region Private members & constants
        public static readonly string layoutTemplatePath = "~/WidgetDesigners/Account/AccountDesigner.ascx";
        public const string scriptReference = "~/WidgetDesigners/Account/AccountDesigner.js";
        #endregion
    }
}
 
